src/lib/dev-environment/dev-environment-cli.js

```diff
@@ -305,26 +305,43 @@ export async function promptForComponent( component: string, allowLocal: boolean
 		const message = `${ messagePrefix }Which version would you like`;
 		const tagChoices = await getTagChoices();
 
+		// Tag strings come back from the api with excess whitespace
+		// This strips the whitespace for matching to the defaultObject.tag
+		const formatted = tagChoices.map( elm => {
+			return elm.trim();
+		} );
+
 		// First tag not: "Pre-Release"
-		const firstNonPreRelease = tagChoices.find( tag => {
-			return ! tag.match( /Pre\-Release/g );
+		const firstNonPreRelease = formatted.find( img => {
+			return ! img.match( /Pre\-Release/g );
 		} );
 
 		// Set initialTagIndex as the first non Pre-Release
-		let initialTagIndex = tagChoices.indexOf( firstNonPreRelease );
+		let initialTagIndex = formatted.indexOf( firstNonPreRelease );
 
 		if ( defaultObject?.tag ) {
-			const defaultTagIndex = tagChoices.indexOf( defaultObject.tag );
+			const defaultTagIndex = formatted.indexOf( defaultObject.tag );
 			if ( defaultTagIndex !== -1 ) {
 				initialTagIndex = defaultTagIndex;
 			}
 		}
 		const selectTag = new Select( {
 			message,
-			choices: tagChoices,
+			choices: formatted,
 			initial: initialTagIndex,
 		} );
-		const tag = await selectTag.run();
+		const option = await selectTag.run();
+
+		// Validate the input
+		// Some of the options are like: '5.7   →  5.7.5'
+		// Extract first occurrence of something that looks like a tag
+		const tagRgx = new RegExp( /(\d+\.\d+(?:\.\d+)?)/ );
```

Comment:
Should we use the meta file instead of parsing lando files?

https://github.com/Automattic/vip/blob/6866b9275a8b3721162076912611c28bd42a8ec9/src/lib/dev-environment/dev-environment-core.js#L239

We already should have all that information in meta file that is used during `update` in a way easier to parse format.

Reply:
Oh sorry this is a wrong place I guess... I am reacting to this from description:

> To fix this I have changed it so that the CLI does not determine the current WP version from the php file, but rather it scrapes the image tag from the .lando.yml file.

Reply:
It took me a sec to get what you are saying but I think you are right. I will make this change and see how it works.

Reply:
In importing `readEnvironmentData` to `dev-environment-cli.js` I realized there is a circular dependency in that `dev-environment-core.js` also imports from `dev-environment-cli.js`

I have looked at the imports on both of these files and also noticed another instance of where I have created a circular dependency with something else.

I need to do some refactoring but before I do that I am going to do this circular dependency anyways because it doesn't actually cause problems it's just grotesque.

This is all just to say that right after this I am going to do a refactor that addresses these circular dependencies.

Reply:
@pschoffer please see the most recent commit. 

I used `readEnvironmentData` to read the environment config, and `prepareLandoEnv` to save changes to the config.

Please let me know if you see anything else that I can fix.